src/gsy_e/models/strategy/new_base_strategy.py

```diff
@@ -0,0 +1,96 @@
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Dict
+
+from gsy_framework.enums import AvailableMarketTypes
+from pendulum import DateTime, duration
+
+from gsy_e.events import EventMixin
+from gsy_e.models.base import AreaBehaviorBase
+from gsy_e.models.strategy.order_updater import (OrderUpdater, OrderUpdaterParameters)
+
+if TYPE_CHECKING:
+    from gsy_e.models.state import StateInterface
+    from gsy_e.models.market import MarketBase
+
+
+class NewStrategyBase(EventMixin, AreaBehaviorBase, ABC):
```

Comment:
I am very open for a better naming for this class. Do you have an idea?

Reply:
I am tempted to suggest `PlanningTradingStrategyBase` in order to indicate that this is the base class that should incorporate both planning and trading algorithms (planning algorithms belong in the EnergyParams classes, which are a member of the strategy class). Or, if we do not want to mention the planning part, `TradingStrategyBase` would also be okay for me.

Reply:
Ok, so you mean that the Baseclass that we used until now `BaseStrategy` only dealt with the trading and with this we distinguish between those. 
I would still vote for TradingStrategyBase.
Eventually (or never, because it is too complicated) we could make the original strategies use this new base blass right?

Reply:
[46102c1](https://github.com/gridsingularity/gsy-e/pull/1565/commits/46102c1a8d2626a8dab189d1e01d0a06e3f2acdc)